090-integrate-bridge-with-metrics-reporter.md

```diff
@@ -0,0 +1,121 @@
+
+```sh
+spec:
+  enableMetrics: true
+```
+
+This is different from how the metrics endpoint is enabled in all the other components deployed by the Cluster Operator.
+The Cluster Operator exposes metrics through the Exporter, which can be configured using the shared `metricsConfig` schema.
+This schema has a type property that only allows the `jmxPrometheusExporter` value, and a reference to a ConfigMap containing its configuration.
+
+```sh
+metricsConfig:
+  type: jmxPrometheusExporter
+  valueFrom:
+    configMapKeyRef:
+      name: my-metrics
+      key: metrics-config.yml
+```
+
+At runtime, the above configuration enables a Java agent that exposes Kafka JMX metrics in Prometheus format through an HTTP endpoint on port 9404.
+Note that the *Exporter* requires `org.apache.kafka.common.metrics.JmxReporter`.
+
+## Motivation
+
+We want to support the *Reporter* as an alternative way of configuring metrics across all Cluster Operator components.
+When deploying any component through the Cluster Operator, we want to provide a consistent user experience, so the Bridge should also support the `metricsConfig` schema. 
+
+## Proposal
+
+The Bridge will support both *Exporter* and *Reporter* as metrics configuration types.
+It will also provide new configurations for the integration with Cluster Operator.
+
+### Metrics configuration types
+
+The Bridge project will be updated to support multiple metrics configurations types.
+A new `bridge.metrics` property will be available within the application.properties configuration file, and will only accept `jmxPrometheusExporter` and `strimziMetricsReporter`.
+Any other value will raise an error and the application will fail to start with an appropriate error message.
+
+When running in standalone mode with `strimziMetricsReporter`, the user will be able to configure any reporter property using the "kafka." prefix.
+The following example will be documented and also provided as a comment in the default application.properties file:
+
+```sh
+bridge.metrics=strimziMetricsReporter
+kafka.metric.reporters=io.strimzi.kafka.metrics.KafkaPrometheusMetricsReporter
+kafka.prometheus.metrics.reporter.listener.enable=false
+kafka.prometheus.metrics.reporter.allowlist=.*
+```
+
+The MetricsReporter class will be updated to also include a new StrimziCollectorRegistry that will work similarly to the JmxCollectorRegistry.
+The StrimziCollectorRegistry will include a reference to PrometheusRegistry.defaultRegistry, which is the same instance used by the *Reporter* to collect metrics.
+
+The kafka_bridge_config_generator.sh script is used to generate the image configuration based on environment variables.
+This script will be also updated to include `bridge.metrics` and related configurations (see the following section).
+
+The Bridge will try to load the Exporter's configuration file from the path specified by the `bridge.metrics.jmx.exporter.config.path` property.
```

Comment:
```suggestion
The Bridge will try to load the configuration file of the *Exporter* from the path specified by the `bridge.metrics.jmx.exporter.config.path` property.
```

Reply:
I'm not clear on this property and the context here? How is it specified and under what circumstances? Is it specified in application.properties if we choose jmxPrometheusExporter?

Reply:
This configuration property will be used by the Cluster Operator to pass the user-provided Exporter's configuration file to the Bridge (see `metricsConfig` schema). The same property can also be leveraged by standalone Bridge users.